### PR TITLE
Fixes #226 Added Shell Sort [Python]

### LINKED ISSUE #226: Add Shell Sort [Python]
I see its missing from the table of algorithms in README.md
https://www.tutorialspoint.com/data_structures_algorithms/shell_sort_algorithm.htm

### PR DESCRIPTION
**Pull request by Enether**

### CODE REVIEW

File: shell_sort/shell_sort.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (aashutoshrathi) [inline shell_sort/shell_sort.py]: remove this blank line

File: shell_sort/shell_sort.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def shell_sort(array):
+    # calculate the gap using Knuth's formula
```
Comment (aashutoshrathi) [inline shell_sort/shell_sort.py]: Add proper docstring

File: shell_sort/shell_sort.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                candidate_idx -= gap
+
+            # insert our element at the hole
+            array[candidate_idx] = val_to_insert
+
+        # decrease gap, math alert
+        gap = (gap - 1) // 3
+
+    return array
+
```
Comment (aashutoshrathi) [inline shell_sort/shell_sort.py]: add new line (as 2 lines needed between methods)

File: shell_sort/shell_sort.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+def main():
+    sample_arr = [1, -312, 4, 12, 3, 17, 2542, 20, 18]
+    print(shell_sort(sample_arr))
+
```
Comment (aashutoshrathi) [inline shell_sort/shell_sort.py]: add new line

File: shell_sort/shell_sort.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+def shell_sort(array):
+    """
+    Sorts the given array of integers using the Shell Sort algorithm
+    """
```
Comment (aashutoshrathi) [inline shell_sort/shell_sort.py]: define parameter array as
` param array : ....`

**Comment by singhpratyush** [comment]
Hello, @Enether! Thanks for the PR. Please go through the Travis build log and make the appropriate changes.

Since we are a bit strict with the coding convention, it may take some time to merge the PR. So please stick with us.

File: shell_sort/shell_sort.py (hunk 5 above)
Comment (Enether) [inline shell_sort/shell_sort.py]: Is this really needed?

Comment (Monal5031) [inline shell_sort/shell_sort.py]: @Enether Recently we have added `PEP8Bear` and `PyLintBear` which checks for these issues, that is why other python codes don't have it, but we'll be fixing it soon.

**Comment by Enether** [comment]
Everything should be okay now, @singhpratyush

File: shell_sort/shell_sort.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+def shell_sort(array: [int]):
```
Comment (singhpratyush) [inline shell_sort/shell_sort.py]: This would be trouble if going with Python 2. I think we should skip the type.

File: shell_sort/shell_sort.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Sorts the given array of integers using the Shell Sort algorithm
+    """
```
Comment (singhpratyush) [inline shell_sort/shell_sort.py]: Please mention the `:param` and `:return` elements in the docstring. Also, just mention the complexities in the docstring.

**Pull request merged**